math/pythagoras.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2020 @author beqakd
+ * @file
+ *
+ * A basic implementation of 3d distance between two point.
+ */
+
+#include <cassert>   // For testing(assert)
+#include <cmath>     // For std::pow(), std::sqrt() and so on
+#include <iostream>  // For io operations
+
+/**
+ * The distance between two points in a three dimensional - 3D - coordinate
+ * system. For this we have struct point where we save coordinates of our points
+ * and later on calculate the distance as mentioned.
+ */
+class Distance {
```

Comment:
i think we can have a `class point` with a member function which takes two points as it's arguments and return the distance between those points, this will make more sense isn't it?

Reply:
Yes, but to take two point i need struct or something right? So, i will remove struct inside class. Will have variables and will call distance method then.

Reply:
Thats why i had struct in the first place in class so i can call method with 2 Points.